docs/src/design/chiplets/kernel_rom.md

```diff
@@ -1,91 +1,101 @@
+
+The kernel ROM enables executing predefined kernel procedures. These procedures are always executed in the root context
+and can only be accessed by a `SYSCALL` operation. The chiplet tracks and enforces correctness of all kernel procedure
+calls as well as maintaining a list of all the procedures defined for the kernel, whether they are executed or not. More
+background about Miden VM execution contexts can be found [here](../../user_docs/assembly/execution_contexts.md).
 
 ## Kernel ROM trace
-The kernel ROM table consists of 6 columns.
+
+The kernel ROM table consists of 5 columns.
+
+> TODO: Update diagram
+> - remove `idx`
+> - duplicate each hash once and set `s_first` = 1 in that row
 
 ![kernel_rom_execution_trace](../../assets/design/chiplets/kernel_rom/kernel_rom_execution_trace.png)
 
 The meaning of columns in the above is as follows:
-- Column $s_0$ specifies whether the value in the row should be included into the chiplets bus $b_{chip}$.
-- $idx$ is a column which starts out at $0$ and must either remain the same or be incremented by $1$ with every row.
-- $r_0, ..., r_3$ are contain the roots of the kernel functions. The values in these columns can change only when the value in the $idx$ column changes. If the $idx$ column remains the same, the values in the $r$ columns must also remain the same.
-
-## Constraints
 
-The following constraints are required to enforce correctness of the kernel ROM trace.
+- Column $s_{first}$ specifies that the start of a block representing a unique kernel procedure hash.
+- $r_0, ..., r_3$ are contain the roots of the kernel functions. The values in these columns can change only
+  when $s_{first}$ is set to 1 in the next row. Otherwise, the values in the $r$ columns remain the same.
 
-For convenience, let's define $\Delta idx = idx' - idx$.
+## Constraints
 
-The $s_0$ column must be binary.
+> Note: the following assumes the ACE chiplet is included in the previous slot, whose documentation will be included
+> in a subsequent PR.
 
-> $$
-s_0^2 - s_0 = 0 \text{ | degree} = 2
-$$
+The following constraints are required to enforce the correctness of the kernel ROM trace.
+_Note: Unless otherwise stated, these constraints should also be multiplied by chiplets module's selector
+flag $chip\_s_4$ for the kernel ROM chiplet, as is true for all constraints in this chiplet._
 
-The value in the $idx$ column must either stay the same or increase by $1$.
+The $s_{first}$ column must be binary.
 
 > $$
-\Delta idx \cdot (1 - \Delta idx) = 0 \text{ | degree} = 2
+s_{first}^2 - s_{first} = 0 \text{ | degree} = 2
 $$
 
-Finally, if the $idx$ column stays the same then the kernel procedure root must not change. This can be achieved by enforcing the following constraint against each of the four procedure root columns:
+$s_{first}$ must be set in the first row of the trace, indicating the start of a new hash which must be included in the set of kernel procedure roots provided by the verifier.
```

Comment:
nit: could we specify precisely the trace we are talking about ? e.g., the kernel ROM chiplet trace, or something like this

Also, I would start by saying the use of the flag and then describe the constraints, against the first row of the chiplet and within the chiplet, e.g, 

$s_first$ indicates the start of a new hash.... 
$s_first$ must be set ...
...